processor/src/main/java/org/mapstruct/ap/internal/model/MappingMethodWithFactory.java

```diff
@@ -0,0 +1,113 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.internal.model;
+
+import java.util.List;
+import java.util.Set;
+
+import org.mapstruct.ap.internal.model.common.Type;
+import org.mapstruct.ap.internal.model.source.Method;
+
+/**
+ * A {@link MappingMethod} that is used by the main mapping methods ({@link BeanMappingMethod},
+ * {@link MapMappingMethod}, {@link IterableMappingMethod} and {@link StreamMappingMethod}
+ *
+ * @author Filip Hrisafov
+ */
+public abstract class MappingMethodWithFactory extends MappingMethod {
```

Comment:
You are right, it is for those 4 Mappings. The factory was something they shared so I used that in the name. However, I am open for suggestions, I am sure it can be better named